specifyweb/frontend/js_src/lib/components/SchemaViewer/TableList.tsx

```diff
@@ -42,46 +42,90 @@ export function SchemaViewerTableList<
       }),
     [sortConfig, unsortedData, applySortConfig]
   );
+  const headerElements = React.useMemo(
+    () =>
+      Object.entries(headers).map(([name, label]) => (
+        <Button.LikeLink onClick={(): void => handleSort(name as FIELD_NAME)}>
+          {label}
+          <SortIndicator fieldName={name} sortConfig={sortConfig} />
+        </Button.LikeLink>
+      )),
+    [headers, handleSort, headerClassName]
+  );
+  return (
+    <GenericSortedDataViewer<DATA>
+      cellClassName={() =>
+        'border border-gray-400 p-2 dark:border-neutral-500 print:p-1'
+      }
+      className="w-fit border border-gray-400 dark:border-neutral-500"
+      data={data}
+      getLink={getLink}
+      headerClassName="border"
+      headerElements={headerElements}
+      headers={headers}
+    />
+  );
+}
+
+export function GenericSortedDataViewer<
+  DATA extends SchemaViewerRow<RR<string, SchemaViewerValue>>
+>({
+  headers,
+  data,
+  getLink,
+  headerElements,
+  className = '',
+  headerClassName = '',
+  cellClassName,
+}: {
+  readonly headers: RR<string, JSX.Element | LocalizedString>;
+  readonly headerElements: RA<JSX.Element>;
```

Comment:
it looks like you are not using headers values at all, only the keys.
so instead of introducing `headerElements` prop, can you use `headers` values instead?

Reply:
Not in some cases, no. headerElements would include button links (to sort)